src/runner/base.ts

```diff
@@ -69,13 +69,44 @@ class BaseRunner {
   }
 
   async terminate() {
+    this.logger.debug(`terminating runner`);
+
     if (['running', 'paused'].includes(this.state)) {
       this.state = 'terminating';
     }
 
-    while (this.state !== 'terminated') {
-      await sleep(5000);
+    let count = 0;
+    while (!this.isTerminated) {
+      if (count === 5) {
```

Comment:
@rrsmith @sugendran should we make this configurable?

Reply:
@johndagostino yes. Ideally we want to line it up with k8s pod termination rules.